robowflex_library/src/io/visualization.cpp

```diff
@@ -56,54 +77,218 @@ void IO::RVIZHelper::updateTrajectories(const std::vector<planning_interface::Mo
+
+        for (const auto &pg : goal.position_constraints)
+        {
+            const auto &pname = pg.link_name;
+
+            Eigen::Affine3d pose = robot_->getLinkTF(pg.header.frame_id);
+            pose.translate(TF::vectorMsgToEigen(pg.target_point_offset));
+
+            for (const auto &primitive :
+                 boost::combine(pg.constraint_region.primitives, pg.constraint_region.primitive_poses))
+            {
+                shape_msgs::SolidPrimitive solid;
+                geometry_msgs::Pose solid_pose;
+                boost::tie(solid, solid_pose) = primitive;
+
+                auto frame = pose * TF::poseMsgToEigen(solid_pose);
+                addGeometryMarker(name, Geometry::makeSolidPrimitive(solid), base_frame, frame, color);
+                addTextMarker(name, name + " " + pname, base_frame, frame * Eigen::Translation3d(0, 0, 0.1),
+                              0.05);
+
+                for (const auto &og : goal.orientation_constraints)
+                {
+                    const auto &oname = og.link_name;
+                    if (oname != pname)
+                        continue;
+
+                    auto q = TF::quaternionMsgToEigen(og.orientation);
+
+                    Eigen::Vector3d scale = {0.1, 0.008, 0.003};
+                    addArrowMarker(name, base_frame, frame * q, color, 1.5 * scale);
+
+                    const auto tolerances = {og.absolute_x_axis_tolerance,  //
+                                             og.absolute_y_axis_tolerance,  //
+                                             og.absolute_z_axis_tolerance};
+                    const auto axes = {Eigen::Vector3d::UnitX(),  //
+                                       Eigen::Vector3d::UnitY(),  //
+                                       Eigen::Vector3d::UnitZ()};
+
+                    for (const auto &angles : boost::combine(tolerances, axes))
+                    {
+                        double value;
+                        Eigen::Vector3d axis;
+                        boost::tie(value, axis) = angles;
+
+                        auto q1 = TF::offsetOrientation(q, axis, value);
+                        addArrowMarker(name, base_frame, frame * q1, color, scale);
+
+                        auto q2 = TF::offsetOrientation(q, axis, -value);
+                        addArrowMarker(name, base_frame, frame * q2, color, scale);
+                    }
+                }
+            }
+
+            // TODO: Implement
+            // for (const auto &mesh :
+            //      boost::combine(pg.constraint_region.meshes, pg.constraint_region.mesh_poses))
+            // {
+            // }
+        }
+    }
+}
+
+void IO::RVIZHelper::removeMarker(const std::string &name)
+{
+    auto markers = markers_.equal_range(name);
+
+    for (auto it = markers.first; it != markers.second; ++it)
+        it->second.action = visualization_msgs::Marker::DELETE;
+}
+
+void IO::RVIZHelper::addMarker(float x, float y, float z)
```

Comment:
Depending on judgment above, either delete, or add a "Deprecated, not adding anything" print here.